File: go_version/internal/models/university.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

const UNIVERSITIES_COLLECTION = "universities"

type University struct {
	ID        primitive.ObjectID `bson:"_id,omitempty"`
	Name      string             `bson:"name,omitempty"`
	Country   string             `bson:"country,omitempty"`
	City      string             `bson:"city,omitempty"`
	IsActive  bool               `bson:"isActive,omitempty"`
	CreatedAt primitive.DateTime `bson:"createdAt,omitempty"`
	UpdatedAt primitive.DateTime `bson:"updatedAt,omitempty"`
	Version   int32              `bson:"__v,omitempty"`
}

File: backend/src/models/University.ts
import mongoose, { Document, Schema } from 'mongoose';

/**
 * University Interface
 */
export interface IUniversity extends Document {
  _id: mongoose.Types.ObjectId;
  name: string;
  country: string;
  city?: string;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

/**
 * University Schema
 */
const UniversitySchema = new Schema<IUniversity>(
  {
    name: {
      type: String,
      required: [true, 'University name is required'],
      unique: true,
      index: true,
      trim: true,
    },
    country: {
      type: String,
      required: [true, 'Country is required'],
      default: 'Palestine',
    },
    city: {
      type: String,
      trim: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

/**
 * Indexes
 * Note: name index is already created by unique: true constraint
 */
UniversitySchema.index({ country: 1 });

const University = mongoose.model<IUniversity>('University', UniversitySchema);

export default University;
